### PR TITLE
[Mailer][DX] Add notice log when email recipients are not defined

### LINKED ISSUE #35805: [Mailer] AbstractTransport silently doing nothing if no recipients set
Today I struggled a bit with sending an email via the Mailer class.

Here is some example code of what tried to do:

```
$email = (new Email())
    ->from('test@test.de')
    ->subject('Subject')
    ->text('Content')
;

$this->mailer->send($email);
```

```
# .env
MAILER_DSN=smtp://localhost:1025
```

The implementation and the mailer configuration looked fine to me, but the email was not send out. There was also no Exception that indicated what I probably did wrong.

So I debugged my code and could find this line in the `AbstractTransport`.

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Mailer/Transport/AbstractTransport.php#L68 

As you can see the `AbstractTransport::send()` method is just returning `null` in case there are no recipients set, which means the transport is failing "silently" without sending the email. Ok, my bad, I forgot to set a recipient. Problem solved.

But I was wondering, if this is intended behaviour? I actually would have expected an Exception telling me that I forgot to set a recipient.

### PR DESCRIPTION
**Pull request by noniagriconomie**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no (can it be considered as yes regarding issues comments?)
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix https://github.com/symfony/symfony/issues/35805 + https://github.com/symfony/symfony/issues/33753
| License       | MIT
| Doc PR        | none

Hi,

Add a log in order to be aware that no email was sent due to bad data
IF the PR is accepted, is the log level `notice` the right one?

thx

### CODE REVIEW

File: src/Symfony/Component/Mailer/Transport/AbstractTransport.php [hunk 1]
```diff
@@ -65,6 +65,8 @@ public function send(RawMessage $message, Envelope $envelope = null): ?SentMessa
         }
 
         if (!$envelope->getRecipients()) {
+            $this->getLogger()->notice('There are no recipients in the mailer envelope. Nothing was sent.');
```
Comment (jschaedl) [inline src/Symfony/Component/Mailer/Transport/AbstractTransport.php]: I would expect a `LogicException` and a hint what I can do to fix the problem.

Comment (noniagriconomie) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php]: @jschaedl i did not know the logic behind the « ?SentMessage » method return signature

that is why i added a log and not thrown ex and remove the return null

Maybe we need to wait other review comment :)

Comment (noniagriconomie) [reply src/Symfony/Component/Mailer/Transport/AbstractTransport.php]: friendly ping @fabpot as you have commented on related issue https://github.com/symfony/symfony/issues/33753

**Comment by fabpot** [comment]
I've fixed the root issue(s) in #36467

**Pull request closed without merging**